Makefile

```diff
@@ -0,0 +1,85 @@
+IMAGE_REPO ?= mirantis/k8s-externalipcontroller
+IMAGE_TAG ?= latest
+
+BUILD_DIR = _output
+GOLANG_IMAGE = golang:1.7
+
+.PHONY: help
+help:
+	@echo "Usage: 'make <target>'"
+	@echo ""
+	@echo "Targets:"
+	@echo "help            - Print this message and exit"
+	@echo "get-deps        - Install project dependencies"
+	@echo "build           - Build ipcontroller binary"
+	@echo "build-image     - Build docker image"
+	@echo "test            - Run all tests"
+	@echo "unit            - Run unit tests"
+	@echo "integration     - Run integration tests"
+	@echo "e2e             - Run e2e tests"
+
+
+.PHONY: get-deps
+get-deps:
+	go get github.com/Masterminds/glide
+	glide install --strip-vendor
+
+
+.PHONY: build
+build: $(BUILD_DIR)/ipcontroller
+
+
+.PHONY: build-image
+build-image: $(BUILD_DIR)/ipcontroller
+	docker build -t $(IMAGE_REPO):$(IMAGE_TAG) .
+
+
+.PHONY: unit
+unit:
+	go test -v ./pkg/...
+
+
+.PHONY: integration
+integration: $(BUILD_DIR)/integration.test .env-prepared
+	sudo $(BUILD_DIR)/integration.test
+
+
+.PHONY: e2e
+e2e: $(BUILD_DIR)/e2e.test .env-prepared
+	sudo $(BUILD_DIR)/e2e.test --master=http://localhost:8888 --testlink=docker0 -ginkgo.v
+
+
+.PHONY: test
+test: unit integration e2e
+
+
+.PHONY: clean
+clean:
+	rm -rf $(BUILD_DIR)
+
+
+$(BUILD_DIR):
+	mkdir -p $(BUILD_DIR)
+
+
+$(BUILD_DIR)/ipcontroller: $(BUILD_DIR)
+	go build -o $@ cmd/ipcontroller.go
```

Comment:
It is not necessary to build it inside container. I'm thinking about adding containerized targets to build binaries inside docker container.